src/asciidoc/reference/packages/withkojiDispatch.adoc

```diff
@@ -51,19 +48,194 @@ const dispatch = new Dispatch({
 });
 ----
 
+== Dispatch properties
+
+[.hcode, id=".clientId", reftext="clientId"]
+=== .clientId
+
+_String_, ID of the currently connected client.
+
+[.hcode, id=".connectedClients", reftext="connectedClients"]
+=== .connectedClients
+
+_Object_, contains IDs of currently connected clients as keys and an object of their user data.
+
+==== Example
+[source,json]
+----
+{
+	"0dc15e36-81b4-442b-be4d-bf590ee73c2a": {
+		username: "test1",
+		lastPing: 1600218622595
+	},
+	"1fe2eb80-57e3-435f-9314-faa74853c34a": {
+		username: "test2",
+		lastPing: 1600218623097
+	}
+}
+----
+
 [.hcode, id=".latency", reftext="latency"]
-=== .latency()
+=== .latency
+
+_Number_, the latency of the last message in milliseconds.
+
+[.hcode, id=".shardName", reftext="shardName"]
+=== .shardName
+
+_String_, name of the shard the client is currently connected to.
+
+[.hcode, id=".userInfo", reftext="userInfo"]
+=== .userInfo
+
+_Object_, the user information of the currently connected client.
+This can be modified using <<#.setUserInfo, .setUserInfo>>.
+
+== Dispatch methods
+
+[.hcode, id=".connect", reftext="connect"]
+=== .connect()
+
+(Async) Connects the dispatch client to an open shard.
+
+==== Example
+
+[source,javascript]
+dispatch.connect();
+
+[.hcode, id=".disconnect", reftext="disconnect"]
+=== .disconnect()
 
-Checks the latency of dispatch messages.
+(Async) Disconnects the dispatch client from the active connection.
+
+==== Example
+
+[source,javascript]
+dispatch.disconnect();
+
+[.hcode, id=".emitEvent", reftext="emitEvent"]
+=== .emitEvent(eventName, payload)
+
+Emits the named event to all clients on the current shard.
+
+==== Parameters
+
+* `eventName` – _String_, name of the event.
+* `payload` – _Any_, data to send with event.
+
+==== Example
+
+[source,javascript]
+dispatch.emitEvent('myEvent', myDataPayload);
+
+[.hcode, id=".info", reftext="info"]
+=== .info()
+
+Gets information about the shard the client is currently connected to.
 
 ==== Returns
 
-Number, the latency of the last message in milliseconds.
+(Async) _ShardInfo_, Object describing the shard the client is currently connected to with shard name and number of connected clients.
 
 ==== Example
 
 [source,javascript]
-const myLatency = dispatch.latency();
+----
+dispatch.info.then((info) => {
+    currentInfo = info;
+});
+----
+
+[.hcode, id=".on", reftext="on"]
+=== .on(eventName, handler)
+
+Sets a listener for a specific event, and runs the handler when the event is dispatched over the shard.
+
+* For Koji custom events see <<#_dispatch_event, DISPATCH_EVENT>>.
+* To send your own events see <<#.emitEvent, .emitEvent>>.
```

Comment:
```suggestion
* To send your own events, see <<.emitEvent>>.
```
These cross-references should work correctly now (that is, pull in the "ref text" automatically). Diddy pushed a fix last week.